lib/install/validate/validator.go

```diff
@@ -549,96 +549,94 @@ func (v *Validator) registries(ctx context.Context, input *data.Data, conf *conf
 	}
 }
 
-func (v *Validator) friendlyRegistryList(registryType string, registryList []url.URL) string {
-	var msg string
-
-	if len(registryList) > 0 {
-		var host string
-		msg = registryType + " = "
-		for i, r := range registryList {
-			host = r.Host
-			if strings.Contains(r.Path, "/") {
-				host = host + r.Path
-			}
-
-			if i == 0 {
-				msg = msg + host
-			} else {
-				msg = msg + ", " + host
-			}
-		}
+func (v *Validator) friendlyRegistryList(registryType string, registryList []string) string {
+	if len(registryList) == 0 {
+		return ""
 	}
 
-	return msg
+	return registryType + " = " + strings.Join(registryList, ", ")
 }
 
 // Validate registries are reachable.  Secure registries that are not specified as insecure are validated with the
 // CA certs passed into vic-machine.
-func (v *Validator) reachableRegistries(ctx context.Context, input *data.Data, pool *x509.CertPool) (insecureRegistries []url.URL, whitelistRegistries []url.URL, err error) {
+func (v *Validator) reachableRegistries(ctx context.Context, input *data.Data, pool *x509.CertPool) (insecureRegistries []string, whitelistRegistries []string, err error) {
 	secureRegistries := input.WhitelistRegistries
 
-	// Remove intersection between insecure registries and whitelist registries from whitelist set so
-	// we can ensure we test the exclusion set with certs
-	var exist bool
-	var idx int
-	var s url.URL
+	// Test insecure registries' reachability
 	for _, r := range input.InsecureRegistries {
-		exist = false
-		for idx, s = range secureRegistries {
-			if r.Host == s.Host {
-				exist = true
+		// Remove intersection between insecure registries and whitelist registries from whitelist set so
+		// we can ensure we test the exclusion set with certs
+		found := false
+		for idx, s := range secureRegistries {
+			if r == s {
+				// remove the insecure registry from list of registries to get validated against certs
+				secureRegistries = append(secureRegistries[:idx], secureRegistries[idx+1:]...)
+				found = true
 				break
 			}
 		}
 
-		// remove the insecure registry from list of registries to get validated against certs
-		if exist {
-			secureRegistries = append(secureRegistries[:idx], secureRegistries[idx+1:]...)
+		if found {
+			continue
```

Comment:
is it continue here?
if there is match in secureRegistry, suppose you still need to validate this address is reachable, though you should not check if the url host match any secure registry again.